cluster-autoscaler/core/scaledown/planner/controller_test.go

```diff
@@ -0,0 +1,177 @@
+			Namespace: "default",
+			UID:       types.UID("apps/v1/namespaces/default/replicasets/rs"),
+		},
+		Spec: appsv1.ReplicaSetSpec{
+			Replicas: proto.Int32(1),
+			Selector: metav1.SetAsLabelSelector(podLabels),
+		},
+		Status: appsv1.ReplicaSetStatus{
+			Replicas: 1,
+		},
+	}
+	rC := &apiv1.ReplicationController{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "rc",
+			Namespace: "default",
+			UID:       types.UID("core/v1/namespaces/default/replicationcontrollers/rc"),
+		},
+		Spec: apiv1.ReplicationControllerSpec{
+			Replicas: proto.Int32(1),
+			Selector: podLabels,
+		},
+		Status: apiv1.ReplicationControllerStatus{
+			Replicas: 0,
+		},
+	}
+	sS := &appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "sset",
+			Namespace: "default",
+			UID:       types.UID("apps/v1/namespaces/default/statefulsets/sset"),
+		},
+		Spec: appsv1.StatefulSetSpec{
+			Replicas: proto.Int32(3),
+			Selector: metav1.SetAsLabelSelector(podLabels),
+		},
+		Status: appsv1.StatefulSetStatus{
+			Replicas: 1,
+		},
+	}
+	rcLister, _ := kube_util.NewTestReplicationControllerLister([]*apiv1.ReplicationController{rC})
+	jobLister, _ := kube_util.NewTestJobLister([]*batchv1.Job{job})
+	rsLister, _ := kube_util.NewTestReplicaSetLister([]*appsv1.ReplicaSet{rs})
+	ssLister, _ := kube_util.NewTestStatefulSetLister([]*appsv1.StatefulSet{sS})
+	listers := kube_util.NewListerRegistry(nil, nil, nil, nil, nil, nil, rcLister, jobLister, rsLister, ssLister)
+	testCases := []struct {
+		name                    string
+		pod                     *apiv1.Pod
+		expectedTargetReplicas  int32
+		expectedCurrentReplicas int32
+		expectErr               bool
+	}{
+		{
+			name:                    "job pod",
+			pod:                     podWithOwnerRef("Job", job.Name, "p1", 1, 2, nil),
+			expectedCurrentReplicas: 1,
+			expectedTargetReplicas:  3,
+		},
+		{
+			name:                    "replica set pod",
+			pod:                     podWithOwnerRef("ReplicaSet", rs.Name, "p1", 1, 2, nil),
+			expectedCurrentReplicas: 1,
+			expectedTargetReplicas:  1,
+		},
+		{
+			name:                    "replica controller pod",
+			pod:                     podWithOwnerRef("ReplicationController", rC.Name, "p1", 1, 2, nil),
+			expectedCurrentReplicas: 0,
+			expectedTargetReplicas:  1,
+		},
+		{
+			name:                    "stateful set pod",
+			pod:                     podWithOwnerRef("StatefulSet", sS.Name, "p1", 1, 2, nil),
+			expectedCurrentReplicas: 0,
+			expectedTargetReplicas:  3,
+		},
+		{
+			name:      "not existing owner ref pod",
+			pod:       podWithOwnerRef("Job", "j", "p1", 1, 1, map[string]string{"app": "notfound"}),
+			expectErr: true,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			c := newControllerCalculator(listers)
+			res, err := c.GetReplicas(tc.pod.OwnerReferences[0], tc.pod)
+			if tc.expectErr {
+				assert.Error(t, err)
+			} else {
+				if res.targetReplicas != tc.expectedTargetReplicas {
```

Comment:
super nit: you could just compare `replicasInfo` here and have a single check.

Reply:
done.